src/util/TunnelIPv6Interface.cpp

```diff
@@ -181,6 +203,19 @@ TunnelIPv6Interface::process(void)
 						break;
 					}
 				}
+			} else if (nlp->nlmsg_type == RTM_NEWLINK || nlp->nlmsg_type == RTM_DELLINK) {
+				struct ifinfomsg *ifinfo = (struct ifinfomsg *)NLMSG_DATA(nlp);
+				const char *ifname = if_indextoname(ifinfo->ifi_index, ifnamebuf);
+				bool isUp, isRunning;
+
+				if ((ifname == NULL) || (get_interface_name() != ifname)) {
+					continue;
+				}
+
+				isUp = ((ifinfo->ifi_flags & IFF_UP) == IFF_UP);
+				isRunning = ((ifinfo->ifi_flags & IFF_RUNNING) == IFF_RUNNING);
+
+				on_link_state_changed(isUp, isRunning);
```

Comment:
The `on_link_state_changed()` seem to be called from a netlink change here. So if the driver itself brings up the interface we don't need to invoke the `on_link_state_changed()` and can rely on netlink event to get the callback, right?